f5/bigip/tm/ltm/policy.py

```diff
@@ -94,6 +96,9 @@ def _modify(self, **patch):
 
         legacy = patch.pop('legacy', False)
         tmos_ver = self._meta_data['bigip']._meta_data['tmos_version']
+        if LooseVersion(tmos_ver) < LooseVersion('12.1.0'):
+            for r in patch.get('rules', []):
+                r.pop('description', None)
```

Comment:
If we are going to make this change, and what it seems this might be one of many such "exceptions", it would be prudent to have this folded into _meta_data, for example _meta_data['optional_parameters'], similarly to:

https://github.com/F5Networks/f5-common-python/blob/development/f5/bigip/resource.py#L839

Also if we silently remove the description from **kwargs how would the caller know something went wrong? If the caller does not know that a certain attribute does not exist in the specific version, we should inform him by raising exception with appropriate method, rather than just removing.

The only exception to such rule is here, and this is due to the fact that we send the entire __dict__ (minus __meta_data) to the device when we do update(), and state/session parameters are read only in specific situations, so the caller need not to be notified, unless the caller attempt to set them to an illegal value:

https://github.com/F5Networks/f5-common-python/blob/development/f5/bigip/tm/ltm/node.py#L95
https://github.com/F5Networks/f5-common-python/blob/development/f5/bigip/tm/ltm/pool.py#L131

We raise if the parameter is illegal value:
https://github.com/F5Networks/f5-common-python/blob/development/f5/bigip/tm/ltm/pool.py#L209

Reply:
The above behavior of update() will be modified once i get a chance, the issue is already opened up for it:
https://github.com/F5Networks/f5-common-python/issues/1063